### ISSUE TITLE
[BUG] [1.4.0.1]  Deferred import bindings break after await statements.

### ISSUE THREAD

**Issue by trusktr** [meteor/meteor#7662]
Found a funky one. Basically, the following code describes the issue, where the `React` variable becomes undefined after the await statement:

``` js
main()
async function main() {
    import React from 'react'
    import startup from 'awaitbox/meteor/startup'

    console.log(' --- React?', React) // defined. All is fine.
    await startup()
    console.log(' --- React?', React) // undefined! Uh oh!
}
```

**Comment by trusktr** [comment]
Here is a reproduction: https://github.com/trusktr/site/tree/meteor-issue-7662

**Comment by benjamn** [comment]
This is a real bug! Thanks for bringing it to my attention.

The best way to solve this would be to turn `reify` into a Babel plugin so that it can run before `babel-plugin-transform-regenerator`, so Regenerator won't accidentally confine the imported variables to the `switch` statement that it generates.

A pretty good solution is just to run the `reify` compiler before any Babel plugins, so that's what I'll work on now.

**Comment by trusktr** [comment]
Wanted to note that for a workaround we can simply define the logic in a function before the await statement, then run the function after the await statement, so the above example becomes:

```js
main()
async function main() {
    import './silence-react'
    import React from 'react'
    import ReactDOM from 'react-dom'

    import startup from 'awaitbox/meteor/startup'

    function logReact() {
        console.log(' --- React?', React) // defined. All is fine.
    }
    
    await startup()

    logReact() // it works
}
```

**Comment by benjamn** [comment]
Yep, there are lots of workarounds. The simplest is to move the `import` declarations out of the `async` function. Here's one version of that:
```js
main()
function main() {
    import './silence-react'
    import React from 'react'
    import ReactDOM from 'react-dom'
    import startup from 'awaitbox/meteor/startup'

    return async function () {
        console.log(' --- React?', React) // defined. All is fine.
        await startup()
        console.log(' --- React?', React) // undefined! Uh oh!
    }();
}
```

**Comment by trusktr** [comment]
I really like deferred imports, I hope they get into spec. What is the likelihood? I'll bring it up in esdiscuss. It can solve circular dependency problems very neatly!

**Comment by benjamn** [comment]
I'd hold off on bringing it up on esdiscuss, since I presented it at the last TC39 meeting, and I'm still working on a revised proposal with some other members.

**Comment by trusktr** [comment]
(Added repro steps to #7646. Are you able to reproduce?)

**Comment by adamgins** [comment]
@benjamn I'm using await/sync to call Stripe... seems like I have a similar issue that await now returns "undefined" when I call Stripe.

Any hints on how to resolve this in the 1.4.2 stream pls?

**Comment by benjamn** [comment]
The `import`-in-`async`-functions bug should be fixed if you run `meteor update --release 1.4.3-beta.0`. Feel free to reopen this issue if that's not the case!
